content/docs/alerting/configuration.md

```diff
@@ -0,0 +1,207 @@
+configuration file defines inhibition rules, notification routing and
+notification receivers.
+
+To view all available command-line flags, run `alertmanager -h`.
+
+Alertmanager can reload its configuration at runtime. If the new configuration
+is not well-formed, the changes will not be applied.
+A configuration reload is triggered by sending a `SIGHUP` to the process.
+
+## Configuration file
+
+To specify which configuration file to load, use the `-config.file` flag.
+
+The file is written in [YAML format](http://en.wikipedia.org/wiki/YAML),
+defined by the scheme described below.
+Brackets indicate that a parameter is optional. For non-list parameters the
+value is set to the specified default.
+
+Generic placeholders are defined as follows:
+
+* `<duration>`: a duration matching the regular expression `[0-9]+[smhdwy]`
+* `<labelname>`: a string matching the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`
+* `<labelvalue>`: a string of unicode characters
+* `<filename>`: a valid path in the current working directory
+* `<boolean>`: a boolean that can take the values `true` or `false`
+* `<string>`: a regular string
+* `<tmpl_string>`: a string which is template-expanded before usage
+
+The other placeholders are specified separately.
+
+A valid example file can be found [here](https://github.com/prometheus/alertmanager/blob/master/doc/examples/simple.yml).
+
+The global configuration specifies parameters that are valid in all other configuration contexts. They also serve as defaults for other configuration
+sections.
+
+
+```
+global:
+  # ResolveTimeout is the time after which an alert is declared resolved
+  # if it has not been updated.
+  [ resolve_timeout: <duration> | default = 5m ]
+
+  # The default SMTP From header field.
+  [ smtp_from: <tmpl_string> ]
+  # The default SMTP smarthost used for sending emails.
+  [ smtp_smarthost: <tmpl_string> ]
+
+  # The API URL to use for Slack notifications.
+  [ slack_api_url: <string> ]
+
+  [ pagerduty_url: <string> | "https://events.pagerduty.com/generic/2010-04-15/create_event.json" ]
+  [ opsgenie_api_host: <string> | "https://api.opsgenie.com/" ]
+
+# Files from which custom notification template definitions are read.
+# The last component may use a wildcard matcher, e.g. 'templates/*.tmpl'.
+templates:
+  [ - <filepath> ... ]
+
+# The root node of the routing tree.
+route: <route>
+
+# A list of inhibition rules.
+inhibit_rules:
+  [ - <inhibit_rule> ... ]
+
+# A list of notification receivers.
+receivers:
+  - <receiver> ...
+```
+
+
+## Route `<route>`
+
+A route block defines a node in a routing tree and its children. Its optional
+configuration parameters are inherited from its parent node if not set.
+
+Every alert enters the routing tree at the configured top-level route, which
+must match all alerts.
+It then traverses the child nodes. If `continue` is set to false, it stops
+after the first matching child. Otherwise it potentially matches all child
```

Comment:
> You said your's is more understandable but didn't give reasons.

It comes from experience, it breaks down as soon as you try to do anything even mildly complicated. As routing rules are business logic that changes with the systems and teams, they tend to end up very complicated with lots of nesting and special cases.

> I couldn't find too many use cases yet for stop-first semantics actually. Most of the time the matchers are mutually exclusive to begin with.

The usual way alert routing ends up is something like:

  * Send copy of all alerts to email
  * specialisation for team X
  * specialisation for subsystem B
  * specialisation for team Y
 *  ...
  * general rule for paging alerts
  * catchall

These rules are not mutually exclusive as you usually want a given alert to only go to one team, and writing them with the all-match is very tricky to do correctly as for every specialisation you add you'd need to change every other rule that could also apply. First match wins is easier to understand and maintain, as you can read it top to bottom and know what happens.

> You didn't address the starvation aspect, which is very real, especially at the top level.

That's what unit tests of your routing rules are for.